### DISCUSSION TITLE
GhostScript Interaction

### DISCUSSION THREAD

**Discussion by GitHubRulesOK** [sumatrapdfreader/sumatrapdf#4539] [General]
SumatraPDF will use any installed GhostScript for **reading** EPS and PS files  by convert to PDF.

Artifex need to change the command line invocations every so often, especially when security is tightened.

Current "New" version 10.04.0 has introduced a new requirement when handling PS files that may (untested) block usage in SumatraPDF 
However in a different common PS writing usage 10.04.0 is not as functional as 10.03.1 and so perhaps worth waiting for the dust to settle?

**Comment by kjk** [comment]
Can you be more specific? What changed in GhostScript? What is the new way of doing the same thing?

**Reply by GitHubRulesOK** [reply]
the 10.04 does not run PS files the same way (I was dealing with a different issue where the PS commands file only works in 10.03 and still does not work in 10.04 perhaps a new bug or restriction) However the release notes mention increased restrictions on running PS files that were not there in 10.03

Unknown impact on SumatraPDF as I am holding at working 10.03.1 x32

> IMPORTANT: In this release (10.04.0) we have be added protection for device selection from PostScript input. This will mean that, by default, only the device specified on the command line will be permitted.
https://ghostscript.readthedocs.io/en/gs10.04.0/News.html?utm_source=ghostscript&utm_medium=website&utm_content=inline-link
